src/zh/sixmh/src/eu/kanade/tachiyomi/extension/zh/sixmh/SixMH.kt

```diff
@@ -1,38 +1,83 @@
-        }
+    override fun popularMangaRequest(page: Int) = GET("$baseUrl/category/order/hits/page/$page", headers)
+    override fun latestUpdatesRequest(page: Int) = GET("$baseUrl/category/order/addtime/page/$page", headers)
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val url = "$baseUrl/index.php/search".toHttpUrl().newBuilder()
+            .addQueryParameter("key", query)
+            .build()
+        return GET(url, headers)
+    }
+
+    override fun simpleNextPageSelector(): String? = null
+    override fun simpleMangaSelector(): String = "div.cy_list_mh ul"
+    override fun simpleMangaFromElement(element: Element): SManga = SManga.create().apply {
+        title = element.selectFirst("li.title > a")!!.text()
+        setUrlWithoutDomain(element.selectFirst("li.title > a")!!.absUrl("href"))
+        thumbnail_url = element.selectFirst("img")?.absUrl("src")
+    }
+
+    // Details
+    override fun mangaDetailsParse(document: Document): SManga = SManga.create().apply {
+        val element = document.selectFirst("div.cy_info")!!
+        title = element.selectFirst("div.cy_title")!!.text()
+        thumbnail_url = element.selectFirst("div.cy_info_cover > a > img.pic")?.absUrl("src")
+        description = element.selectFirst("div.cy_desc #comic-description")?.text()
+
+        val infoElements = element.select("div.cy_xinxi")
+        author = infoElements[0].selectFirst("span:first-child > a")?.text()
+        status = parseStatus(infoElements[0].selectFirst("span:nth-child(2)")?.text())
+        genre = infoElements[1].selectFirst("span:first-child > a")?.text()
     }
 
-    override fun getMangaUrl(manga: SManga) = "https://m.liumanhua.com" + manga.url
-    override fun getChapterUrl(chapter: SChapter) = "https://m.liumanhua.com" + chapter.url
+    // Chapters
+    override fun chapterListSelector(): String = "ul#mh-chapter-list-ol-0 li.chapter__item"
+    override fun chapterFromElement(element: Element): SChapter = SChapter.create().apply {
+        setUrlWithoutDomain(element.selectFirst("a")!!.absUrl("href"))
+        name = element.selectFirst("a > p")!!.text()
+    }
 
+    // Pages
     override fun pageListParse(response: Response): List<Page> {
-        val encodedData = dataRegex.find(response.body.string())?.groupValues?.get(1) ?: ""
-        val cid = response.request.url.pathSegments.last().removeSuffix(".html").toIntOrNull() ?: 0
-        val decodedData = decodeData(encodedData, cid)
-        val images = json.decodeFromString<List<Image>>(decodedData)
-        return images.mapIndexed { index, image -> Page(index, imageUrl = image.url) }
+        val body = response.body.string()
+        val encodedData = paramsRegex.find(body)?.groupValues?.get(1) ?: ""
+        val decodedData = decodeData(encodedData)
+
+        val images = json.decodeFromString<Data>(decodedData).images
```

Comment:
```suggestion
        val images = decodedData.parseAs<Data>().images
```
need to import it too, and with this, remove the `json` from above too